### PR TITLE
Clear 1.7.0.0.sql

### PR DESCRIPTION
**Pull request by martinfojtik**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | merge some queries |
| Type? | improvement |
| Category? | CO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? |  |
| How to test? | update prestashop or use a sql validator |

### CODE REVIEW

**Review by mickaelandrieu — APPROVED** [review]

**Review by firstred — APPROVED** [review]

File: install-dev/upgrade/sql/1.7.0.0.sql
Comment (aleeks) [inline install-dev/upgrade/sql/1.7.0.0.sql]: It was twice, but I think it's a mistake nope ?

File: install-dev/upgrade/sql/1.7.0.0.sql
Comment (aleeks) [inline install-dev/upgrade/sql/1.7.0.0.sql]: Why not with others on the top ?

**Review by aleeks — CHANGES_REQUESTED** [review]
`UPDATE`PREFIX_quick_access`SET`link`= "product/new" WHERE`link`= "index.php?controller=AdminProducts&addproduct";`

 **need to be changed into :** 

`UPDATE`PREFIX_quick_access`SET`link`= "`**index.php/product/new**`" WHERE`link`= "index.php?controller=AdminProducts&addproduct";`

It's an error, (not you), see install-dev/data/xml/quick_access.xml

After this change, I make a test with the update file! :smile:

File: install-dev/upgrade/sql/1.7.0.0.sql
Comment (martinfojtik) [inline install-dev/upgrade/sql/1.7.0.0.sql]: sure, i moved `PS_ORDER_RECALCULATE_SHIPPING` to the top.

**Review by aleeks — APPROVED** [review]

**Comment by aleeks** [comment]
Hello @martinfojtik 
It's ok for me, I'm waiting travis test, and I make a test from 1.6.1.7!
Very good job !

File: install-dev/upgrade/sql/1.7.0.0.sql
Comment (aleeks) [inline install-dev/upgrade/sql/1.7.0.0.sql]: Same for ALTER TABLE `PREFIX_product_attribute` ADD `isbn` VARCHAR( 13 ) NULL DEFAULT NULL;..

Need ALTER TABLE `PREFIX_product_attribute` CHANGE `available_date` `available_date` DATE NULL DEFAULT NULL;

And can you report to db_structure ??
available_date date NOT NULL DEFAULT '0000-00-00', line 1512

And can you make the changement for : PREFIX_product_attribute_shop ??
Just add in the file :
ALTER TABLE `PREFIX_product_attribute_shop` CHANGE `available_date` `available_date` DATE NULL DEFAULT NULL;

And report it on db_structure : 
`available_date` date NOT NULL DEFAULT '0000-00-00', line 1532...
To prevent error for future..

File: install-dev/upgrade/sql/1.7.0.0.sql
Comment (aleeks) [inline install-dev/upgrade/sql/1.7.0.0.sql]: Can you put NOW() insteed '0000-00-00 00:00:00' 

sql

File: install-dev/upgrade/sql/1.7.0.0.sql
Comment (aleeks) [inline install-dev/upgrade/sql/1.7.0.0.sql]: For this 4 requests, can you add ` char ? on name table.

File: install-dev/upgrade/sql/1.7.0.0.sql
Comment (aleeks) [inline install-dev/upgrade/sql/1.7.0.0.sql]: Can you add a IF EXISTS to all DROP TABLE .. to prevent errors please ?..

File: install-dev/upgrade/sql/1.7.0.0.sql
Comment (aleeks) [inline install-dev/upgrade/sql/1.7.0.0.sql]: Line 62, same as last_connection_date, can you add : 

ALTER TABLE `PREFIX_product` CHANGE `available_date` `available_date` DATE NULL DEFAULT NULL;
ALTER TABLE `PREFIX_product_shop` CHANGE `available_date` `available_date` DATE NULL DEFAULT NULL;

and report it on db_structure.sql lines : 1433 & 1478


sql3

File: install-dev/upgrade/sql/1.7.0.0.sql
Comment (aleeks) [inline install-dev/upgrade/sql/1.7.0.0.sql]: Line 76 (i can point this), same as last_connection_date : 

ALTER TABLE `PREFIX_product_attribute_shop` CHANGE `available_date` `available_date` DATE NULL DEFAULT NULL;
ALTER TABLE `PREFIX_product_attribute` CHANGE `available_date` `available_date` DATE NULL DEFAULT NULL;

and report it on db_structure.sql lines : 1512 & 1532

File: install-dev/upgrade/sql/1.7.0.0.sql
Comment (aleeks) [inline install-dev/upgrade/sql/1.7.0.0.sql]: Can you add before, because it make an error : 
ALTER TABLE `PREFIX_employee` CHANGE `last_connection_date` `last_connection_date` DATE NULL DEFAULT NULL;

sql_2

**Review by aleeks — CHANGES_REQUESTED** [review]
Hello, can you fix this little things please ?
I was testing this sql file and I had some mistake!

If i'm not clear, say me!

Thank you for your job, really.

**Review by jocel1 — COMMENTED** [review]
Would be interesting to merge this PR after merging #6531 .
I think the issues reported by @aleeks are fixed by PR #6531 (a SET SESSION sql_mode='' present in DbPDO.php but not in DbMySQLi.php)

File: install-dev/upgrade/sql/1.7.0.0.sql
Comment (martinfojtik) [inline install-dev/upgrade/sql/1.7.0.0.sql]: sure...

File: install-dev/upgrade/sql/1.7.0.0.sql
Comment (martinfojtik) [inline install-dev/upgrade/sql/1.7.0.0.sql]: i'm not sure what you want to do. You want to change `varchar `to `char`?

File: install-dev/upgrade/sql/1.7.0.0.sql
Comment (martinfojtik) [inline install-dev/upgrade/sql/1.7.0.0.sql]: yes, and i moved some `Alter` queries closer together

File: install-dev/upgrade/sql/1.7.0.0.sql
Comment (aleeks) [inline install-dev/upgrade/sql/1.7.0.0.sql]: Hello @martinfojtik 
Just add  **'PREFIX_employee'** :smile: but with " ` " i can make it because, it's interpreted like code ahah

**Comment by julienbourdeau** [comment]
Great job @martinfojtik! It will make our lives much easier :)

Thank you very much @aleeks for testing and reviewing in details.

**Review by aleeks — APPROVED** [review]

**Comment by aleeks** [comment]
Executed without error !
Really good job @martinfojtik !!

I put this PR in WIP in order to **not merge it**, waiting for #6531!

**Comment by julienbourdeau** [comment]
I think it will be easier to rebase @jocel1's PR against this one. Let's merge it!

**Comment by martinfojtik** [comment]
Thanks @aleeks for testing my changes 👍

**Pull request merged**
